src/Libplanet.Action/ActionEvaluator.cs

```diff
@@ -267,7 +268,8 @@ IActionContext CreateActionContext(IWorld newPrevState)
                     txs: inputContext.Txs,
                     previousState: newPrevState,
                     randomSeed: inputContext.RandomSeed,
-                    gasLimit: inputContext.GasLimit());
+                    gasLimit: inputContext.GasLimit(),
+                    evidences: inputContext.Evidences);
```

Comment:
How do we collect evidence in action?

Reply:
Custom evidence can be added to the evidence pool of a `BlockChain` from the external layer of a library or application. However, the blockchain object cannot be accessed from within an action. Therefore, currently, there is no functionality to directly collect evidence within an action. If evidence can be collected based on failed transactions in a `BlockRender` and added to the `Blockchain`, it would be possible.

Reply:
Then why need `Evidences` in `InputContext`?

Reply:
It is not needed now, but this property is required to execute the slash code in DPoS.

Reply:
👌